src/StringHelper.php

```diff
@@ -577,4 +577,54 @@ static function (string $key) use ($delimiter, $escapeCharacter): string {
             $result
         );
     }
+
+    /**
+     * Strip Unicode whitespace (with property White_Space=yes) or other characters from the beginning and end of a string.
+     * Input string and pattern are treated as UTF-8.
+     *
+     * @see https://en.wikipedia.org/wiki/Whitespace_character#Unicode
+     * @see https://www.php.net/manual/function.preg-replace
+     *
+     * @param string $string The input string.
+     * @param string $pattern PCRE regex pattern to search for, as a string. Quote $pattern if it contains special regular expression characters.
+     * @see https://www.php.net/manual/function.preg-quote.php
+     *
+     * @return string
+     */
+    public static function trim(string $string, string $pattern = "\pC\pZ"): string
+    {
+        if (!preg_match('##u', $pattern)) {
+            throw new InvalidArgumentException('Pattern must be valid UTF-8 string.');
+        }
+
+        return preg_replace("#^[$pattern]+|[$pattern]+$#uD", '', $string);
+    }
+
+    /**
+     * Strip Unicode whitespace (with property White_Space=yes) or other characters from the beginning of a string
+     *
+     * @see self::trim()
+     */
+    public static function ltrim(string $string, string $pattern = "\pC\pZ"): string
```

Comment:
```suggestion
    public static function ltrim(string $string, string $pattern = self::DEFAULT_WHITESPACE_PATTERN): string
```
Could you please extract the value to a constant and reuse it?